requests/__init__.py

```diff
@@ -86,6 +87,25 @@ def check_compatibility(urllib3_version, chardet_version):
 except ImportError:
     pass
 
+# Check for older version of pyOpenSSL
+def check_pyopenssl(cryptography_version, pyopenssl_version):
+    # Only applies to pyOpenSSL
+    if sys.modules['OpenSSL'].__title__ == 'pyOpenSSL':
+        # Warning only applies for Cryptography < 1.3.4
+        if list(map(int, cryptography_version.split('.'))) < [1, 3, 4]:
+            # Warning only applies for pyOpenSSL < 16.0.0
+            assert list(map(int, cryptography_version.split('.'))) < [16, 0, 0]
+
+# Check cryptography version
+if 'OpenSSL' in sys.modules:
```

Comment:
I also don't think we actually need the check to be this complicated. We're already checking for `pyOpenSSL` with our `inject_pyopenssl_into_urllib3` above. A simple helper function that checks cryptography's version meets or exceeds our threshold (supposedly 1.3.4), and otherwise displays a warning, should be sufficient.